rusk-uplink/src/hosted.rs

```diff
@@ -0,0 +1,225 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use crate::RawTransaction;
+pub use crate::{
+    ArchiveError, ContractId, ContractState, Query, RawQuery, ReturnValue,
+    Transaction,
+};
+use bytecheck::CheckBytes;
+use microkelvin::{OffsetLen, StoreRef, StoreSerializer};
+use rkyv::validation::validators::DefaultValidator;
+use rkyv::{Archive, Deserialize, Serialize};
+
+const BUFFER_SIZE_LIMIT: usize = 1024 * 16;
+
+// declare available host-calls
+pub mod external {
```

Comment:
Shouldn't we make it clear that we're using pointers and not references here? It's allowed by the compiler, but that doesn't mean it's clear. Instead of `&u8` maybe we should use `*const u8`.